forem/forem
# Add "only display when logged out" as an option on Navigation Links

## What type of PR is this? (check all applicable)


## Description

Previously we had a single check box when creating navigation links: "Show only when logged in", which was either checked or unchecked (show the link to everyone). This PR introduces a third option: show only to logged out users. As such, we now present the options as a radio group.

I've also added a new DB column for this, since it didn't conform to the old boolean value. Definitely keen for input on whether there's a better way to do this - the way I've approached it, we'll have a redundant `display_only_when_signed_in` column that needs cleaned up after this merges.

## Related Tickets & Documents



- Related Issue #
- Closes https://github.com/forem/forem-internal-eng/issues/494

## QA Instructions, Screenshots, Recordings

- Before running this branch, make sure you have some existing navigation links set to "only show to logged in users" - you can configure nav links in `/admin/customization/navigation_links`
- Check out this branch locally and run `bin/setup && bin/startup` 
- Make sure that the nav links you set to 'only show to logged in users' still are not visible when signed out & that they are visible when signed in
- Visit `/admin/customization/navigation_links` and create and/or edit nav links, experimenting with the options 'show to all users' / 'only logged in' / 'only logged out' - check that they appear/disappear in the nav sidebar when expected

### UI accessibility concerns?

Nothing much - the radio buttons are grouped and labelled in a fieldset to help communicate their purpose

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

After this merges there will be a redundant column `display_only_when_signed_in` that can be removed in a follow-up (unless I get some suggestions on this PR of a more efficient way to handle this)
